app/src/main/java/com/android/ubclaunchpad/driver/session/SessionInfoActivity.java

```diff
@@ -185,4 +215,68 @@ public void onCancelled(DatabaseError databaseError) {
                     }
                 });
     }
+
+
+    /**
+     * Retrieve all drivers and passengers from the current session
+     *
+     * @return ArrayList of all Users in the current session
+     */
+    private void setDriverPassengers() {
+
+        final List<User> users = new ArrayList<>();
+
+        DatabaseReference usersReference = FirebaseUtils.getDatabase().child("Users");
+        // Getting reference to the users in Firebase
+        usersReference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(final DataSnapshot dataSnapshotUsers) {
+                // Getting reference to the drivers in current session
+                session.addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot dataSnapshot) {
+
+                        DataSnapshot dataSnapshotDrivers = dataSnapshot.child("drivers");
+                        System.out.println(dataSnapshotDrivers.getChildrenCount());
+
+                        for (DataSnapshot driver : dataSnapshotDrivers.getChildren()) {
+                            String driverUid = driver.getValue(String.class);
+
+                            User uDriver = dataSnapshotUsers.child(driverUid).getValue(User.class);
+                            uDriver.setIsDriver(true);
+                            uDriver.setUserUid(driverUid);
+                            System.out.println(uDriver.name);
+                            users.add(uDriver);
+                        }
+                        DataSnapshot dataSnapshotPassengers = dataSnapshot.child("passengers");
+                        for (DataSnapshot passenger : dataSnapshotPassengers.getChildren()) {
+                            String passengerUid = passenger.getValue(String.class);
+                            User uPassenger = dataSnapshotUsers.child(passengerUid).getValue(User.class);
+                            uPassenger.setUserUid(passengerUid);
+                            users.add(uPassenger);
+
+                        }
+
+                        String locStr = dataSnapshot.child("location").getValue(String.class);
+                        LatLng location = StringUtils.stringToLatLng(locStr);
+
+                        System.out.println(users.size());
+                        FindBestRouteAlgorithm algorithm = new FindBestRouteAlgorithm(location);
+                        Map<String, Object> driverPassengers = algorithm.findBestRoute(users);
+                        System.out.println(driverPassengers.size());
+
+                        session.child("driverPassengers").updateChildren(driverPassengers);
```

Comment:
We should probably add "driverPassengers" to StringUtils :) Also "Users", "passengers" and "drivers" are already in StringUtils so you could use those instead ...